test/models/fact_value_test.rb

```diff
@@ -67,6 +67,21 @@ def setup
     refute_empty results
   end
 
+  test 'should return empty search results for ILIKE operator on host' do
+    results = FactValue.search_for("host ~ abc")
+    assert_empty results
+  end
+
+  test 'should return search results for ILIKE operator on hostgroup' do
+    host = FactoryGirl.create(:host, :hostgroup => FactoryGirl.create(:hostgroup, :name => 'samplehgp'))
+    FactoryGirl.create(:fact_value, :value => '2.6.9',:host => host,
+                       :fact_name => FactoryGirl.create(:fact_name, :name => 'kernelversion'))
+    search_text = host.hostgroup.to_label
+    results = FactValue.search_for("host.hostgroup ~ #{search_text}")
```

Comment:
I would try searching on just a part of the hostgroup name to make sure the ILIKE works correctly, something like `ampl`.